### PR TITLE
add write docstring action

### PR DESCRIPTION
**Pull request by shenchucheng**
- Adding docstrings for Python code
- Supporting Sphinx/Google/Numpy styles
- Preserving the actual text of the original code associated with each node using libcst

### CODE REVIEW

File: metagpt/actions/write_docstring.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+PYTHON_DOCSTRING_SYSTEM = '''### Requirements
+1. Add docstrings to the given code following the {style} style.
+2. Replace the function body with an Ellipsis object(...) to reduce output.
+3. If the types are already annotated, there is no need to include them in the docstring.
+4. Extract only class, function or the docstrings for the module parts from the given Python code, avoiding any other text.
+
+### Input Example
+```python
+def function_with_pep484_type_annotations(param1: int) -> bool:
+    return isinstanc(param1, int)
```
Comment (better629) [inline metagpt/actions/write_docstring.py:18]: wrong word

File: metagpt/actions/write_docstring.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+class ExampleError(Exception):
+    def __init__(self, msg: str):
+        self.msg = msg
+```
+
+### Output Example
+```python{example}```
```
Comment (better629) [inline metagpt/actions/write_docstring.py:26]: python
{example}
?

File: metagpt/utils/common.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def parse_blocks(cls, text: str):
 
     @classmethod
     def parse_code(cls, text: str, lang: str = "") -> str:
-        pattern = rf'```{lang}.*?\s+(.*?)```'
+        pattern = rf'```{lang}.*?\s+(.*)```'
```
Comment (better629) [inline metagpt/utils/common.py:52]: make sure it won't influence existing code parsing strategy

Comment (shenchucheng) [reply metagpt/utils/common.py:52]: This function `OutputParser.parse_code` is only refered here.

File: metagpt/actions/write_docstring.py (hunk 2 above)
Comment (shenchucheng) [reply metagpt/actions/write_docstring.py:26]: The example is one of `PYTHON_DOCSTRING_EXAMPLE_GOOGLE`, `PYTHON_DOCSTRING_EXAMPLE_NUMPY`, or `PYTHON_DOCSTRING_EXAMPLE_SPHINX` code snippets, which depend on the style you use. These code snippets all begin and end with `\n`.

File: metagpt/actions/write_docstring.py (hunk 1 above)
Comment (shenchucheng) [reply metagpt/actions/write_docstring.py]: fixxed

File: metagpt/actions/write_docstring.py (hunk 2 above)
Comment (better629) [reply metagpt/actions/write_docstring.py:26]: although example snippets seperate with `\n`, the `python\n{example}` format is more clear.

Comment (shenchucheng) [reply metagpt/actions/write_docstring.py:26]: OK, I will remove the newline characters from the example snippets.

**Comment by better629** [comment]
LGTM

**Comment by voidking** [comment]
1. Add usage example like: 
```bash
export PYTHONPATH=$PYTHONPATH:$(pwd)
python metagpt/actions/write_docstring.py workspace/cli_2048_game/cli_2048_game/game.py --overwrite
```

2. Sometimes parse code error.

**Comment by voidking** [comment]
LGTM

**Pull request merged**